collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java

```diff
@@ -108,20 +97,30 @@ public void cancelLongPress() {
         stringAnswer.cancelLongPress();
     }
 
+    @Override
     protected void onDetachedFromWindow() {
         super.onDetachedFromWindow();
         if (customTabHelper.getServiceConnection() != null) {
             getContext().unbindService(customTabHelper.getServiceConnection());
         }
     }
 
+    private boolean isUrlEmpty(TextView stringAnswer) {
```

Comment:
Even lower please :) both are called only by onButtonClick() so it looks better if we have them below. Sorry for being picky but if you are rebuilding the widget it's good do fix even such small things.

Reply:
Oh no that's okay! It's good if we have all these things fixed.